include/zephyr/posix/unistd.h

```diff
@@ -55,6 +55,153 @@ pid_t getpid(void);
+#define __z_posix_sysconf_SC_BC_STRING_MAX		  (-1L)
+#define __z_posix_sysconf_SC_CHILD_MAX			  (-1L)
+#define __z_posix_sysconf_SC_CLK_TCK			  CONFIG_SYS_CLOCK_TICKS_PER_SEC
+#define __z_posix_sysconf_SC_COLL_WEIGHTS_MAX		  (-1L)
+#define __z_posix_sysconf_SC_DELAYTIMER_MAX		  (-1L)
+#define __z_posix_sysconf_SC_EXPR_NEST_MAX		  (-1L)
+#define __z_posix_sysconf_SC_HOST_NAME_MAX		  (-1L)
+#define __z_posix_sysconf_SC_IOV_MAX			  (-1L)
+#define __z_posix_sysconf_SC_LINE_MAX			  (-1L)
+#define __z_posix_sysconf_SC_LOGIN_NAME_MAX		  (-1L)
+#define __z_posix_sysconf_SC_NGROUPS_MAX		  (-1L)
+#define __z_posix_sysconf_SC_GETGR_R_SIZE_MAX		  (-1L)
+#define __z_posix_sysconf_SC_GETPW_R_SIZE_MAX		  (-1L)
+#define __z_posix_sysconf_SC_MQ_OPEN_MAX		  (-1L)
+#define __z_posix_sysconf_SC_MQ_PRIO_MAX		  (-1L)
+#define __z_posix_sysconf_SC_OPEN_MAX			  (-1L)
+#define __z_posix_sysconf_SC_PAGE_SIZE			  (-1L)
+#define __z_posix_sysconf_SC_PAGESIZE			  (-1L)
+#define __z_posix_sysconf_SC_THREAD_DESTRUCTOR_ITERATIONS (-1L)
+#define __z_posix_sysconf_SC_THREAD_KEYS_MAX		  COND_CODE_1(CONFIG_PTHREAD_IPC,	\
+								(CONFIG_MAX_PTHREAD_KEY_COUNT), \
+								(-1L))
+#define __z_posix_sysconf_SC_THREAD_STACK_MIN		  PTHREAD_STACK_MIN
+#define __z_posix_sysconf_SC_THREAD_THREADS_MAX		  (-1L)
+#define __z_posix_sysconf_SC_RE_DUP_MAX			  (-1L)
+#define __z_posix_sysconf_SC_RTSIG_MAX			  (-1L)
+#define __z_posix_sysconf_SC_SEM_NSEMS_MAX		  (-1L)
+#define __z_posix_sysconf_SC_SEM_VALUE_MAX		  COND_CODE_1(CONFIG_PTHREAD_IPC,     \
+								      (CONFIG_SEM_VALUE_MAX), \
+								      (-1L))
+#define __z_posix_sysconf_SC_SIGQUEUE_MAX		  (-1L)
+#define __z_posix_sysconf_SC_STREAM_MAX			  (-1L)
+#define __z_posix_sysconf_SC_SYMLOOP_MAX		  (-1L)
+#define __z_posix_sysconf_SC_TIMER_MAX			  CONFIG_MAX_TIMER_COUNT
+#define __z_posix_sysconf_SC_TTY_NAME_MAX		  (-1L)
+#define __z_posix_sysconf_SC_TZNAME_MAX			  (-1L)
+#define __z_posix_sysconf_SC_ADVISORY_INFO		  (-1L)
+#define __z_posix_sysconf_SC_BARRIERS			  COND_CODE_1(CONFIG_PTHREAD_IPC,	  \
+								      (SUPPORTED_POSIX1_VERSION), \
+								      (-1L))
+#define __z_posix_sysconf_SC_ASYNCHRONOUS_IO		  (-1L)
+#define __z_posix_sysconf_SC_CLOCK_SELECTION		  (-1L)
+#define __z_posix_sysconf_SC_CPUTIME			  (-1L)
+#define __z_posix_sysconf_SC_FSYNC			  (-1L)
+#define __z_posix_sysconf_SC_IPV6			  (-1L)
+#define __z_posix_sysconf_SC_JOB_CONTRO			  (-1L)
+#define __z_posix_sysconf_SC_MAPPED_FILES		  (-1L)
+#define __z_posix_sysconf_SC_MEMLOCK			  (-1L)
+#define __z_posix_sysconf_SC_MEMLOCK_RANGE		  (-1L)
+#define __z_posix_sysconf_SC_MEMORY_PROTECTION		  (-1L)
+#define __z_posix_sysconf_SC_MESSAGE_PASSING		  (-1L)
+#define __z_posix_sysconf_SC_MONOTONIC_CLOCK		  (-1L)
+#define __z_posix_sysconf_SC_PRIORITIZED_IO		  (-1L)
+#define __z_posix_sysconf_SC_PRIORITY_SCHEDULING	  (-1L)
+#define __z_posix_sysconf_SC_RAW_SOCKETS		  (-1L)
+#define __z_posix_sysconf_SC_READER_WRITER_LOCKS	  (-1L)
```

Comment:
This would probably be more accurate
```suggestion
#define __z_posix_sysconf_SC_READER_WRITER_LOCKS		  COND_CODE_1(CONFIG_PTHREAD_IPC, (_POSIX_VERSION), (-1L))
```

although realistically, it should probably be a separate Kconfig option - maybe `POSIX_RWLOCK`? Feel free to add that as a separate commit before this commit in the same PR. Otherwise, we can do it later in a separate PR.